packages/react-scripts/config/env.js

```diff
@@ -24,8 +24,12 @@ if (!NODE_ENV) {
   );
 }
 
+const DOTENV = process.env.DOTENV;
```

Comment:
I've used similar approach, but changed line #20 to

```
const DOT_ENV = process.env.REACT_APP_DOT_ENV || process.env.NODE_ENV;
```

and used `DOT_ENV` instead of `NODE_ENV` so that all files are using same ENV name.

Reply:
Can we use this approach to build a production optimized bundle with staging specific configs?

Reply:
Yes, the only difference is with your change is in the dot env files to be used, while keeping NODE_ENV equal to production. With your change:
`.env.staging.local, .env.staging, .env.production.local, .env.production, .env.local, .env`
with mine:
`.env.staging.local, .env.staging, .env.local, .env`

Probably I like first one more as it allows optional override of production env files via staging files instead of full replacement 👍

Reply:
@fantactuka Hi,That's great thinking and I was modified my `env.js`,In my `.env.staging` file have a variable `REACT_APP_ROOT_API` , I  want  override `.env.development` , But it's not, Because The `.env.development`'s behind `.env.staging` , so  finally I modify:   
```
const DOT_ENV = process.env.DOT_ENV || NODE_ENV
....

`${paths.dotenv}.${DOT_ENV}.local`,

`${paths.dotenv}.${DOT_ENV}`,

```  
My question is: 
  In normally we hope the custom env file  whole overrides standard env file, Isn't it？Please tell me why are you do?Thank you so much.

Reply:
@moersing Actually I won't hope the custom env file whole overrides standard env file. I just wanted to build the same optimized codes, but deploy to different environments with individual env variables like GA code. In my situation, appending the settings and overwrite it if necessary may be more productive. 😄

Reply:
Well, I agree your thinking, But I think about another question , Just mentioned at above , How determine which `REACT_APP_ROOT_API ` is we need? According to your opinion , ummm.... we put a variable in the `.env.staging` named `isStage` then judge in code ？Though that no problem but I feeling not so good.....:wink:, Anyway  thank you for your replied.